calendar-bundle/src/Resources/contao/classes/Calendar.php

```diff
@@ -144,20 +150,18 @@ protected function generateFiles($arrFeed)
 					continue;
 				}
 
+				$objParent = $this->getPageWithDetails($jumpTo);
```

Comment:
why is this done outside the `if (!isset($arrUrls[$jumpTo]))` call?

Reply:
The correct parent page always needs to be set, regardless of whether an URL to that parent page was already generated or not. Otherwise `$GLOBALS['objPage']` might be the parent page of a previous news item, rather than the current one.